extensions/shared/blocks/cover/cover-media-placeholder.js

```diff
@@ -9,13 +8,13 @@ import { useContext, useCallback } from '@wordpress/element';
 /**
  * Internal dependencies
  */
-import { isUpgradable, isVideoFile } from './utils';
+import { isVideoFile } from './utils';
 import { CoverMediaContext } from './components';
 
 export default createHigherOrderComponent(
 	CoreMediaPlaceholder => props => {
 		const { name } = useBlockEditContext();
-		if ( ! name || ! isUpgradable( name ) ) {
+		if ( 'core/cover' !== name ) {
 			return <CoreMediaPlaceholder { ...props } />;
 		}
```

Comment:
This checking is redundant since it's begin controlled by _isUpgradable()_, in the **addVideoUploadPlanCheck** function.

I see it's k removing it.

Reply:
From what I understand, adding a filter to `editor.MediaPlaceholder` adds it to all instances, even when the current block is not `core/cover`. Is that not the case?

Reply:
yeah, it's added to all blocks but the block type control is handled by the `isUpgradable()` function too. So, in theory, the media placeholder will be extended only for core/cover beyond of this control.

Reply:
[here](https://github.com/Automattic/jetpack/blob/6572c94e3f07c30a57cc522a405f745663be1e50/extensions/shared/blocks/cover/utils.js#L58)

Reply:
I `console.log`'d the name variable on my sandbox and it *does* add the filter to all instances of `editor.MediaPlaceholder`. I'm afraid we'll still need that check

Reply:
I see, the block type checking in the _isUpgradable()_ function doesn't make sense, because once the function is bound to the hook, it executes for all blocks.

Reply:
I think does make sense there because we only want to add the filter if `core/cover` gets registered. If it never gets registered we don't need to add the notice.